Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ShowViewshedFromPointOnMapView: View {
+    /// The point on the screen where the user tapped.
+    @State private var tapScreenPoint: Point?
+    
+    /// The current geoprocessing status.
+    @State private var geoprocessingInProgress = false
+    
+    /// The data model for the sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        MapView(map: model.map, graphicsOverlays: [model.inputGraphicsOverlay, model.resultGraphicsOverlay])
+            .onSingleTapGesture { _, tapPoint in
+                tapScreenPoint = tapPoint
+            }
+            .overlay(alignment: .center) {
+                // Sets indication when geoprocessingInProgress is true.
+                if geoprocessingInProgress {
+                    ProgressView("Geoprocessing in progress…")
+                        .padding()
+                        .background(.ultraThinMaterial)
+                        .cornerRadius(10)
+                        .shadow(radius: 50)
+                }
+            }
+            .task(id: tapScreenPoint) {
```

Comment:
If you tap the map again while the geoprocessing job is running, you'll see that the translucent hud is no longer appearing. Try to fix it 😉 and think about creative ways to actually derive the in progress status from the job.